source/_posts/2020-04-14-the-future-of-yaml.markdown

```diff
@@ -0,0 +1,233 @@
+This is great! Sharing ideas, providing inspiration for all of us!
+
+There is a great downside to this, which is privacy and security. There are a
+lot of things we don’t want to share. For example, our passwords,
+sensitive (personal) information or (historical) data.
+
+A bit more technical, but integrations using OAuth2 as the authentication method
+to integrate with, for example, a service like Somfy don’t need to store the
+username/password at all.
+
+## Both YAML & UI
+
+Home Assistant is moving towards a better separation of YAML configuration
+versus configuration done in the UI. This is already partly showing in things
+like zones, Lovelace and the recently introduced “Helpers”
+(also known as the `input_*` integrations in YAML).
+
+Zones can be configured via the UI and via YAML (even at the same time!).
+Both configuration sources will be loaded by Home Assistant when considering
+the configured zones and both can be changed on the fly without restarting
+Home Assistant.
+
+For the future, Home Assistant will be moving towards that concept more
+and more. This allows anybody to use the method they prefer.
+
+In these cases, YAML support has been extended and improved by adding features
+like reloading and by removing parts that mess with your YAML files.
+
+## We are still an open-source project
+
+Home Assistant is an open-source project and relies on contributors adding
+those amazing integrations, maintaining them and extending their functionality
+to make them even more capable.
+
+Those contributors do this in their free spare time, for which we all are
+eternally grateful. It is their work that enables Home Assistant to do what it
+can right now. It is what automates your home.
+
+So what about integrations that remove YAML support?
+
+Some contributors have decided to remove the YAML support to reduce their
+maintenance and support burden. The amount of energy that needs to be put in
+(to maintain both capabilities) can be too much and is complex. We have to
+understand and accept that. If we do not do that, a contributor could simply
+stop contributing.
+
+Unfortunately, such a move creates breaking changes and often leads to a few
+pretty de-motivating comments, towards the contributor and the project in
+general. This is harmful to everybody, as the contributors get demotivated or,
+even worse, don’t want to implement new features or create a breaking change.
+
+This halts our project goals, slows down innovation and induces the risk of
+losing contributors and maintainers. In the end, leads to a greater loss for
+everybody.
+
+## The future of YAML
+
+So, with all of the above set context: What is the future of YAML?
+
+As of today, an ADR (Architectural Decision Record) has been approved that
+states:
+
+- Any new integration that communicates with devices and/or services, must use
+  configuration via the UI. Configuration via YAML is only allowed in very rare
+  cases, which will be determined on a case by case basis.
```

Comment:
Should we give an example of a rare case?

Reply:
There is no strict ruling on that. Giving an example, loosen the ADR, which I think we should not do.